src/M6Web/Bundle/LogBridgeBundle/Config/Filter.php

```diff
@@ -40,130 +31,67 @@ public function __construct($name)
         $this->options = [];
     }
 
-    /**
-     * getName
-     *
-     * @return string
-     */
-    public function getName()
+    public function getName(): string
     {
         return $this->name;
     }
 
-    /**
-     * setRoute
-     *
-     * @param string $route Route
-     *
-     * @return Filter
-     */
-    public function setRoute($route)
+    public function setRoute(?string $route): self
     {
         $this->route = $route;
 
         return $this;
     }
 
-    /**
-     * getRoute
-     *
-     * @return string
-     */
-    public function getRoute()
+    public function getRoute(): ?string
     {
         return $this->route;
     }
 
-    /**
-     * setMethod
-     *
-     * @param mixed $method Method
-     *
-     * @return Filter
-     */
-    public function setMethod($method)
+    public function setMethod(mixed $method): self
     {
         $this->method = $method;
 
         return $this;
     }
 
-    /**
-     * getMethod
-     *
-     * @return mixed
-     */
-    public function getMethod()
+    public function getMethod(): mixed
     {
         return $this->method;
     }
 
-    /**
-     * setStatus
-     *
-     * @param mixed $status Http code status
-     *
-     * @return Filter
-     */
-    public function setStatus($status)
+    public function setStatus(mixed $status): self
     {
         $this->status = $status;
 
         return $this;
     }
 
-    /**
-     * getStatus
-     *
-     * @return mixed
-     */
-    public function getStatus()
+    public function getStatus(): mixed
     {
         return $this->status;
     }
 
-    /**
-     * setLevel
-     *
-     * @param mixed $level Log level
-     *
-     * @return Filter
-     */
-    public function setLevel($level)
+    public function setLevel(?string $level): self
```

Comment:
why not `mixed` like the others?

Reply:
Maybe, because we know the $level [type](https://github.com/BedrockStreaming/LogBridgeBundle/pull/68/files#diff-999d6f0b9190c37b81242fb5cf9dacfd247f391c5487b24b1d5781c032fb2a68R20) :blush:

Reply:
`setLevel` is only used in this file `M6Web/Bundle/LogBridgeBundle/Config/FilterParser.php` and the method looks like this 
```
protected function parseLevel(Filter $filter, ?string $level): void
    {
        if (!is_string($level) && $level !== null) {
            throw new ParseException(sprintf('Unrecognized value "%s" from level parameter', $level));
        }
        // ...
    }
```

I figured "why would I allow mixed when I know exactly what's being used ?"

Reply:
this $level type is also something that we changed to `?string` in this PR

so my question about `mixed` is also true for the $level type and for the `getLevel()` function

Reply:
@Shoormax ok, but the signature of `parseLevel()` is also something that we change in this PR, it currently accepts something else than a string

![Capture d’écran 2022-05-23 à 10 43 25](https://user-images.githubusercontent.com/2571084/169780326-00608753-e940-497d-a4f0-828efb3c6734.png)


before this PR, if we pass an `int` as $level to `parseLevel()`, it will throw a `ParseException`

after this PR, if we pass an `int` as $level to `parseLevel()`, it will throw a `TypeError`

this may not be a big deal, but it will break `parseLevel()` callers if they only `catch (ParseException)` or `catch (\Exception)` instead of `catch (\Throwable)`

Reply:
Yep, you're right, nice catch there & thanks for the explanation, I'll change it back to `mixed`